scr/components/App.jsx

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useState } from "react";
+
+import { Key } from "./Key";
+import { Button } from "./Button";
+
+import { Api } from "../utils/Api";
+
+export const App = () => {
+  const [keys, setKeys] = useState([]);
+  const [isLoading, setLoadingState] = useState(false);
+  const [countUsedKeys, setCountUsedKeys] = useState(0);
+
+  const toggleLoading = useCallback(() => {
+    setLoadingState(!isLoading);
+  }, [isLoading]);
+
+  const incrementUsedKeys = () => {
+    setCountUsedKeys((prevValue) => prevValue++);
+  }
+
+  const decrementUsedKeys = useCallback(() => {
+    setCountUsedKeys((prevValue) => prevValue--);
+  }, []);
+
+  const addKey = useCallback(async () => {
+    toggleLoading();
+
+    const key = await Api.generateKey();
+    setKeys((prevKeys) => prevKeys.push(key));
+
+    toggleLoading();
+  }, [toggleLoading]);
+
+  const removeKey = useCallback((value) => {
+    setKeys((prevKeys) => prevKeys.filter((key) => key !== value));
+  }, []);
+
+  return (
+    <main>
+      <div>
+        <h3>Всего ключей: {keys.length}</h3>
+        <h3>Использовано текущих ключей: {countUsedKeys}</h3>
+      </div>
+
+      {!keys.length && <div>Список ключей пуст</div>}
+
+      {keys.length && (
```

Comment:
Условия обратные. Для чистоты лучше вынести в тернарный оператор. Это исключит ситуацию одновременного отображения списка и текста, что список пуст, если условие будет изменено только в одном месте
```suggestion
      {keys.length ? (
        <div className="keys">
        ...
        </div>
    )  : <div>Список ключей пуст</div>}
```